FirebaseAuth/Tests/Unit/FIROAuthProviderTests.m

```diff
@@ -171,6 +177,38 @@ @interface FIROAuthProviderTests : XCTestCase
 
 @end
 
+#pragma mark - FIRKFakeAppCheckResult
+
+/// A fake appCheckResult used for dependency injection during testing.
+@interface FIRKFakeAppCheckResult : NSObject <FIRAppCheckTokenResultInterop>
+@property(nonatomic) NSString *token;
+@property(nonatomic, nullable) NSError *error;
+@end
+
+@implementation FIRKFakeAppCheckResult
+
+@end
+
+#pragma mark - FIRFakeAppCheck
+
+/// A fake appCheck used for dependency injection during testing.
+@interface FIRKFakeAppCheck : NSObject <FIRAppCheckInterop>
+@property(nonatomic, nonnull, readwrite, copy) NSString *tokenDidChangeNotificationName;
+@property(nonatomic, nonnull, readwrite, copy) NSString *notificationAppNameKey;
+@property(nonatomic, nonnull, readwrite, copy) NSString *notificationTokenKey;
+@end
+
+@implementation FIRKFakeAppCheck
+
+- (void)getTokenForcingRefresh:(BOOL)forcingRefresh
+                    completion:(nonnull FIRAppCheckTokenHandlerInterop)completion {
+  FIRKFakeAppCheckResult *fakeAppCheckResult = [[FIRKFakeAppCheckResult alloc] init];
+  fakeAppCheckResult.token = kFakeAppCheckToken;
+  completion(fakeAppCheckResult);
+}
+
+@end
```

Comment:
To avoid having to redefine this multiple times (with slight edits to the name of the class), can you pull out this `FIRFakeAppCheck` and `FIRFakeAppCheckResult` implementation into its own file? i.e. in FirebaseAuth/Tests/Unit/FIRFakeAppCheck.m or something like that. (The existing implementation in FIRAuthBackendRPCImplementationTests.m can also be extracted. That way, we avoid code duplication and symbol conflict issues.

Reply:
done, thx